pkg/trait/prometheus.go

```diff
@@ -121,10 +121,13 @@ func (t *prometheusTrait) Apply(e *Environment) (err error) {
 	options := []string{strconv.Itoa(t.Port), path.Join(prometheusJmxExporterConfigMountPath, prometheusJmxExporterConfigFileName)}
 	container.Args = append(container.Args, "-javaagent:dependencies/io.prometheus.jmx.jmx_prometheus_javaagent-0.3.1.jar="+strings.Join(options, ":"))
 
-	// Add the container port
-	containerPort := t.getContainerPort()
-	container.Ports = append(container.Ports, *containerPort)
-	condition.Message = fmt.Sprintf("%s(%s/%d)", container.Name, containerPort.Name, containerPort.ContainerPort)
+	// Skip declaring the Prometheus port when Knative is enabled,
+	// as only one container port is supported.
+	if e.DetermineProfile() != v1.TraitProfileKnative {
```

Comment:
Not sure about this. Knative profile means that knative is present in the cluster, while the integration is not automatically exposed as Knative service (e.g. think to timer to log). Should we check the actual controller type?

Reply:
Ah you're right, let me rework it!

Reply:
I've updated the logic to rely on the controller strategy used for the integration rather than the trait profile. Thanks!